src/SoundOptionsModal.test.js

```diff
@@ -0,0 +1,114 @@
+                {
+                    onCancel: mockOnCancel,
+                    onChangeSoundOptions: mockOnSave
+                },
+                props
+            )
+        ),
+        {
+            wrappingComponent: IntlProvider,
+            wrappingComponentProps: {
+                locale: 'en',
+                defaultLocale: 'en',
+                messages: messages.en
+            }
+        }
+    );
+
+    return {
+        wrapper,
+        mockOnCancel,
+        mockOnSave
+    };
+}
+
+function getAudioToggleSwitches(wrapper) {
+    return wrapper.find(ToggleSwitch);
+}
+
+function getCancelButton(wrapper) {
+    return wrapper.find('.ModalWithFooter__secondaryButton');
+}
+
+function getSaveButton(wrapper) {
+    return wrapper.find('.ModalWithFooter__primaryButton');
+}
+
+describe('When rendering audio toggles', () => {
+    test('Three toggle switches should render in following order: "All Sounds", "Musical Sounds" and "Audio Announcements"', () => {
+        expect.assertions(7);
+        const { wrapper } = createMountSoundOptionsModal();
+        const audioToggleSwitches = getAudioToggleSwitches(wrapper);
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(true);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(true);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(true);
+    });
+    test('Changing audioEnabled, announcementsEnabled and sonificationEnabled properties to false should turn off corresponding toggle siwtches', () => {
+        expect.assertions(7);
+        const { wrapper } = createMountSoundOptionsModal({ audioEnabled: false, announcementsEnabled: false, sonificationEnabled: false });
+        const audioToggleSwitches = getAudioToggleSwitches(wrapper);
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(false);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(false);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(false);
+    })
+});
+
+describe('When interacting with footer buttons', () => {
+    test('Pressing cancel button should close the modal without changing toggle states', () => {
+        expect.assertions(2);
+        const { wrapper, mockOnCancel } = createMountSoundOptionsModal();
+        const cancelButton = getCancelButton(wrapper);
+        cancelButton.at(0).simulate('click');
```

Comment:
For this test to match the description, you'd need to:

1. Change something
2. Confirm that the change isn't persisted.

You could also just rename it to reflect what it tests, i..e that the cancel button is correctly wired up.